control/controlhttp/client_js.go

```diff
@@ -28,9 +28,15 @@ func Dial(ctx context.Context, addr string, machineKey key.MachinePrivate, contr
 	if err != nil {
 		return nil, err
 	}
+	wsScheme := "wss"
+	wsHost := host
+	if host == "localhost" {
+		wsScheme = "ws"
+		wsHost = net.JoinHostPort(host, addr)
```

Comment:
Line 27 makes this super confusing... addr changes from a host:port to just a port. Rename the variable on line 27?

Reply:
Yeah, not sure what I was thinking there. Renamed that to `port`